packages/0x.js/src/utils/order_state_utils.ts

```diff
@@ -96,6 +103,77 @@ export class OrderStateUtils {
+                   transferrableMakerFeeTokenAmount.gte(remainingMakerFeeAmount)) {
+            return remainingMakerAmount;
+        } else {
+            return this.calculatePartiallyFillableMakerTokenAmount(
+              transferrableMakerTokenAmount, transferrableMakerFeeTokenAmount, remainingMakerAmount,
+              remainingMakerFeeAmount, totalMakerAmount, makerFeeAmount, makerTokenAddress,
+              zrxTokenAddress);
+        }
+    }
+    private calculatePooledFillableMakerTokenAmount(transferrableMakerTokenAmount: BigNumber,
+                                                    transferrableMakerFeeTokenAmount: BigNumber,
+                                                    remainingMakerAmount: BigNumber,
+                                                    remainingMakerFeeAmount: BigNumber,
+                                                    totalMakerAmount: BigNumber, makerFeeAmount: BigNumber,
+                                                    makerTokenAddress: string, zrxTokenAddress: string): BigNumber {
+        if (transferrableMakerTokenAmount.plus(transferrableMakerFeeTokenAmount).gte(
+            remainingMakerAmount.plus(remainingMakerFeeAmount))) {
+            return remainingMakerAmount;
+        } else {
+            return this.calculatePartiallyFillableMakerTokenAmount(
+              transferrableMakerTokenAmount, transferrableMakerFeeTokenAmount, remainingMakerAmount,
+              remainingMakerFeeAmount, totalMakerAmount, makerFeeAmount, makerTokenAddress,
+              zrxTokenAddress);
+        }
+    }
+    private calculatePartiallyFillableMakerTokenAmount(transferrableMakerTokenAmount: BigNumber,
+                                                       transferrableMakerFeeTokenAmount: BigNumber,
+                                                       remainingMakerAmount: BigNumber,
+                                                       remainingMakerFeeAmount: BigNumber,
+                                                       totalMakerAmount: BigNumber, makerFeeAmount: BigNumber,
+                                                       makerTokenAddress: string, zrxTokenAddress: string): BigNumber {
+        const orderToFeeRatio = totalMakerAmount.dividedToIntegerBy(makerFeeAmount);
+        const fillableTimesInFeeToken = BigNumber.min(transferrableMakerFeeTokenAmount, remainingMakerFeeAmount);
+        let fillableTimesInMakerToken = transferrableMakerTokenAmount.dividedToIntegerBy(orderToFeeRatio);
+        if (makerTokenAddress === zrxTokenAddress) {
+            const totalFeeTokenPool = transferrableMakerTokenAmount.plus(transferrableMakerFeeTokenAmount);
+            fillableTimesInMakerToken = totalFeeTokenPool.dividedToIntegerBy(
+                                                             orderToFeeRatio.plus(
+                                                                 ZeroEx.toBaseUnitAmount(new BigNumber(1), 18)));
```

Comment:
This is incorrect. We need to only add the 1 ZRX baseUnit to the part of the ratio that corresponds to the ZRX fee.